modin/numpy/arr.py

```diff
@@ -1434,41 +1529,308 @@ def sum(
+        if truthy_where or out is not None:
             return fix_dtypes_and_determine_return(
-                result, new_ndim, dtype, out, where is not False
+                result, new_ndim, dtype, out, truthy_where
             )
         else:
             return (
                 numpy.zeros_like(array(_query_compiler=result, _ndim=new_ndim))
                 + initial
             )
 
+    def all(self, axis=None, out=None, keepdims=None, *, where=True):
+        check_kwargs(keepdims=keepdims, where=where)
+        truthy_where = bool(where)
+        target = where.where(self, True) if isinstance(where, array) else self
+        if self._ndim == 1:
+            if axis == 1:
+                raise numpy.AxisError(1, 1)
+            result = target._query_compiler.all(axis=0)
+            if keepdims:
+                if out is not None and out.shape != (1,):
+                    raise ValueError(
+                        f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
+                    )
+                if truthy_where or out is not None:
+                    return fix_dtypes_and_determine_return(
+                        result, 1, bool, out, truthy_where
+                    )
+                else:
+                    return array([True], dtype=bool)
+            result = result.to_numpy()[0, 0] if truthy_where else True
+            return result if where else True
+        if axis is None:
+            result = target._query_compiler.all(axis=1).all(axis=0)
+            if keepdims:
+                if out is not None and out.shape != (1, 1):
+                    raise ValueError(
+                        f"operand was set up as a reduction along axis 0, but the length of the axis is {out.shape[0]} (it has to be 1)"
+                    )
+                if truthy_where or out is not None:
+                    return fix_dtypes_and_determine_return(
+                        array(numpy.array([[result]], dtype=bool))._query_compiler,
+                        2,
+                        bool,
+                        out,
+                        truthy_where,
+                    )
+                else:
+                    return array([[True]], dtype=bool)
+            return result.to_numpy()[0, 0] if truthy_where else True
+        result = target._query_compiler.all(axis=axis)
+        new_ndim = self._ndim - 1 if not keepdims else self._ndim
+        if new_ndim == 0:
+            result = result.to_numpy()[0, 0]
+            return result if truthy_where else True
+        if not keepdims and axis != 1:
+            result = result.transpose()
+        if truthy_where or out is not None:
+            return fix_dtypes_and_determine_return(
+                result, new_ndim, bool, out, truthy_where
+            )
+        else:
+            return numpy.ones_like(array(_query_compiler=result, _ndim=new_ndim))
```

Comment:
Shouldn't we throw a dtype conversion on this, so that its an array full of Trues (rather than of 1's, which are semantically equivalent)?

Reply:
`ones_like` inherits the type of its argument array, and `result` here is already bools after the call to the query compiler method.

https://numpy.org/doc/stable/reference/generated/numpy.ones_like.html
> Return an array of ones with the same shape and type as a given array.